shared/common-adapters/checkbox.native.js

```diff
@@ -0,0 +1,109 @@
+/* @flow */
+
+import React, {Component} from 'react'
+import {View, TouchableWithoutFeedback, Animated, Easing} from 'react-native'
+import Text from './text'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import type {Props} from './checkbox'
+
+const checkedOffset = 14
+
+type State = {
+  left: any
+}
+
+class Checkbox extends Component {
+  props: Props;
+  state: State;
+
+  _getOffset (props: Props): number {
+    return props.checked ? checkedOffset : 0
+  }
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {left: new Animated.Value(this._getOffset(props))}
```

Comment:
this uses the animated framework, so read up on that if you've never seen this before